dashboard-server/queries/responses/response.test.js

```diff
@@ -57,34 +73,357 @@ describe('Response queries', () => {
+
+      const { email, surveyName, after } = mockData;
+
+      const responses = await Response.all(email, surveyName, after);
+
+      responses.items[0].should.eql({
+        token: 'MTk3MC0wMS0wMSAwMDowMDowMCswMDowMA==',
+        answers: [
+          {
+            parent_surveyid: survey.id,
+            parent_shortcode: '101',
+            surveyid: survey.id,
+            flowid: '100001',
+            userid: '127',
+            question_ref: 'ref',
+            question_idx: '10',
+            question_text: 'text',
+            response: '{ "text": "last" }',
+            timestamp: timestamps[1],
+            metadata: null,
+            pageid: null,
+            translated_response: null,
+          },
+          {
+            parent_surveyid: survey.id,
+            parent_shortcode: '101',
+            surveyid: survey.id,
+            flowid: '100004',
+            userid: '127',
+            question_ref: 'ref',
+            question_idx: '10',
+            question_text: 'text',
+            response: '{ "text": "first" }',
+            timestamp: timestamps[2],
+            metadata: null,
+            pageid: null,
+            translated_response: null,
+          },
+          {
+            parent_surveyid: survey.id,
+            parent_shortcode: '101',
+            surveyid: survey.id,
+            flowid: '100004',
+            userid: '128',
+            question_ref: 'ref',
+            question_idx: '10',
+            question_text: 'text',
+            response: '{ "text": "first" }',
+            timestamp: timestamps[2],
+            metadata: null,
+            pageid: null,
+            translated_response: null,
+          },
+          {
+            parent_surveyid: survey.id,
+            parent_shortcode: '202',
+            surveyid: survey.id,
+            flowid: '100005',
+            userid: '126',
+            question_ref: 'ref',
+            question_idx: '10',
+            question_text: 'text',
+            response: '{ "text": "first" }',
+            timestamp: timestamps[3],
+            metadata: null,
+            pageid: null,
+            translated_response: null,
+          },
+        ],
+      });
+      describe('userNotFound', () => {
+        it('should return no responses if the user email is not found', async () => {
+          const userNotFound = await Response.all(
+            'userdoesntexist@vlab.com',
+            surveyName,
+            after,
+          );
+
+          const answers = userNotFound.items[0].answers;
+          answers.length.should.equal(0);
+        });
+      });
+
+      it('should return a response if the user email is found', async () => {
+        const userFound = await Response.all(email, surveyName, after);
+
+        const answers = userFound.items[0].answers;
+        answers.length.should.equal(4);
+      });
+
+      describe('surveyNotFound', () => {
+        it('should return no responses if the survey name is not found', async () => {
```

Comment:
I'm not convinced it should return an empty array if the survey isn't found, as opposed to a 404!